portfolio/src/test/java/com/google/sps/TextValidatorTest.java

```diff
@@ -0,0 +1,105 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** */
+@RunWith(JUnit4.class)
+public final class TextValidatorTest {
+
+  private static final TextValidator validator = TextValidator.getInstance();
+
+  private static final String VALID1 = "This is a common type of text.";
+
+  private static final String VALID2 = "This test uses 3 numbers: 1, 2, 3";
+
+  private static final String VALID3 = "This test contains special chars! : , ( ? .";
+
+  private static final String INVALID1 = "<script> alert(1); </script>";
+
+  private static final String INVALID2 = "<img=x onerror=alert(1)>";
+
+  private static final String INVALID3 = "; return 0;";
+
+  private static final String INVALID4 = "name=Andrei or 1=1 --;";
+
+  @Test
+  public void commonValidComment() {
+    boolean actual = validator.validate(VALID1, TextValidator.InputType.COMMENT);
+    boolean expected = true;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void numbersComment() {
+    boolean actual = validator.validate(VALID2, TextValidator.InputType.COMMENT);
+    boolean expected = true;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void symbolsComment() {
+    boolean actual = validator.validate(VALID3, TextValidator.InputType.COMMENT);
+    boolean expected = true;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void firstXssInjection() {
+    boolean actual = validator.validate(INVALID1, TextValidator.InputType.COMMENT);;
+    boolean expected = false;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void secondXssInjection() {
+    boolean actual = validator.validate(INVALID2, TextValidator.InputType.COMMENT);
+    boolean expected = false;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void otherInjection() {
+    boolean actual = validator.validate(INVALID3, TextValidator.InputType.COMMENT);
+    boolean expected = false;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void sqlInjection() {
+    boolean actual = validator.validate(INVALID4, TextValidator.InputType.COMMENT);
+    boolean expected = false;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void longComment() {
+    String comment = new String();
+    for (int i = 0; i <= TextValidator.COMMENT_LENGTH_LIMIT + 1; ++i) {
+      comment = comment + 'a';
+    }
+    System.out.println(comment.length());
+    boolean actual = validator.validate(comment, TextValidator.InputType.COMMENT);
+    boolean expected = false;
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void longName() {
+    String name = new String();
+    for (int i = 0; i <= TextValidator.NAME_LENGTH_LIMIT + 1; ++i) {
```

Comment:
If the "name length limit" was a class member, you could have saved yourself from writing for-loops and instead could have written

```
String name = "012345";
TextValidator limit6 = new TextValidator();
limit6.setNameLengthLimit(6);
assertTrue(limit6.validate(name));
```

By choosing to hard-code the validator properties into constants, you now have to write the tests around your choice in the implementation. I'm highlighting this here because this is one of the great benefits of test-driven development that the needs for the test might have changed the validator interface to make it easier to write the tests.